presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloMetadata.java

```diff
@@ -0,0 +1,544 @@
+
+import com.facebook.presto.accumulo.metadata.AccumuloTable;
+import com.facebook.presto.accumulo.model.AccumuloColumnHandle;
+import com.facebook.presto.accumulo.model.AccumuloTableHandle;
+import com.facebook.presto.accumulo.model.AccumuloTableLayoutHandle;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.ConnectorInsertTableHandle;
+import com.facebook.presto.spi.ConnectorMetadata;
+import com.facebook.presto.spi.ConnectorOutputTableHandle;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.ConnectorTableLayout;
+import com.facebook.presto.spi.ConnectorTableLayoutHandle;
+import com.facebook.presto.spi.ConnectorTableLayoutResult;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.ConnectorViewDefinition;
+import com.facebook.presto.spi.Constraint;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.SchemaTablePrefix;
+import com.facebook.presto.spi.TableNotFoundException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.slice.Slice;
+
+import javax.inject.Inject;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.accumulo.AccumuloErrorCode.ACCUMULO_TABLE_EXISTS;
+import static com.facebook.presto.accumulo.Types.checkType;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Presto metadata provider for Accumulo. Responsible for creating/dropping/listing tables, schemas,
+ * columns, all sorts of goodness. Heavily leverages {@link AccumuloClient}
+ */
+public class AccumuloMetadata
+        implements ConnectorMetadata
+{
+    private final String connectorId;
+    private final AccumuloClient client;
+
+    @Inject
+    public AccumuloMetadata(AccumuloConnectorId connectorId, AccumuloClient client)
+    {
+        this.connectorId = requireNonNull(connectorId, "connectorId is null").toString();
+        this.client = requireNonNull(client, "client is null");
+    }
+
+    /**
+     * Begins the process of creating a table with the given metadata. This process of
+     * begin/commit/rollback is used for CTAS statements.
+     *
+     * @param session Current client session
+     * @param tableMetadata Table metadata to create
+     * @return Output table handle
+     */
+    @Override
+    public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session,
+            ConnectorTableMetadata tableMetadata)
+    {
+        SchemaTableName stName = tableMetadata.getTable();
+        AccumuloTable table = client.createTable(tableMetadata);
```

Comment:
This is not thread safe since `AccumuloClient.createTable` calls `validateCreateTable` which calls `validateExternalTable` which uses `Indexer`, which is not thread safe.
If you used new Connector API (with transactions) you could benefit from single-thread per transaction guarantee it gives you.